core/consensus/grandpa/impl/authority_manager_impl.cpp

```diff
@@ -194,6 +195,8 @@ namespace kagome::consensus::grandpa {
 
     BOOST_ASSERT_MSG(root_ != nullptr, "The root must be initialized by now");
 
+    fixKusamaHardFork(*root_);
```

Comment:
This should be run only for kusama-network. Please, add check of that (i.e. by genesis block hash).